### PR TITLE
feat: skip next subscription action with hooks

### PR DESCRIPTION
**Pull request by kibo-sahilarora**

### CODE REVIEW

File: components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx [hunk 1]
```diff
@@ -75,6 +79,37 @@ const SubscriptionItemList = (props: SubscriptionItemListProps) => {
   const { getProductLink } = uiHelpers()
   const { t } = useTranslation('common')
 
+  const { showModal } = useModalContext()
+  const { showSnackbar } = useSnackbarContext()
+  const skipNextSubscription = useSkipNextSubscriptionMutation()
+
+  const skipNextOrderConfirmation = (subscriptionId: string) => {
```
Comment (TriptiVermaXP) [inline components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx]: rename to handleSkipShipment or handleSkipNextSubscription

File: components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx [hunk 2]
```diff
@@ -75,6 +79,37 @@ const SubscriptionItemList = (props: SubscriptionItemListProps) => {
...
+    showModal({
+      Component: ConfirmationDialog,
+      props: {
+        onConfirm: () => callSkipSubscriptionMutation(subscriptionId),
```
Comment (TriptiVermaXP) [inline components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx]: rename to confirmSkipShipment or confirmSkipNextSubscription

File: __test__/e2e/integration/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    const skipShipmentButton = screen.getByRole('button', {
+      name: /skip-shipment/i,
+    })
+
+    await user.click(skipShipmentButton)
+
+    const confirmationText = screen.getByText(/skip-next-subscription-confirmation/i)
+    expect(confirmationText).toBeVisible()
+    expect(screen.getByRole('button', { name: 'close' })).toBeVisible()
```
Comment (TriptiVermaXP) [inline __test__/e2e/integration/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.spec.tsx:40]: this should be unit test not integration.

File: components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx (hunk 1 above)
Comment (kibo-sahilarora) [reply components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx]: Done

File: components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx (hunk 2 above)
Comment (kibo-sahilarora) [reply components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx]: Done.

**Review by TriptiVermaXP — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]

**Review by kibo-sushant — APPROVED** [review]

**Pull request merged**